Sources_v3/Models/CurrentUser.swift

```diff
@@ -49,6 +49,9 @@ public class _CurrentChatUser<ExtraData: UserExtraData>: _ChatUser<ExtraData> {
     /// A set of users flagged by the user.
     public let flaggedUsers: Set<_ChatUser<ExtraData>>
     
+    /// A set of message ids flagged by the user.
+    public let flaggedMessageIDs: Set<MessageId>
```

Comment:
We cannot have `flaggedMessage: Set<_ChatMessage>` because the current-user's model has **ExtraData: UserExtraData** while the message has **ExtraData: ExtraDataTypes**. 

I suggest to go with `flaggedMessageIDs` and change it to `flaggedMessage` when we update the `CurrentChatUser`:
1. to be a `struct` as other models are
2. to have have `user: _ChatUser` via the composition + @dynamicMemberLookup for the underlaying values
3. to change **ExtraData: UserExtraData** to **ExtraData: ExtraDataTypes**

That's mostly done in c4ad5b7ef383ac126d658bd11d3c41dde0a53f9b

Reply:
This is a very nice comment that should've been in the PR description before you asked someone for a review 😅

Reply:
BTW - I think the approach with ids is better and exposing the model objects directly was a mistake that slipped through the review.

Reply:
Gotcha 👍 Will also create a ticket for the `flaggedUsers` and `mutedUsers` fix

Reply:
Here it is [CIS-408](https://stream-io.atlassian.net/browse/CIS-408)

Reply:
Thanks 👍